src/app/services/user.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+
+import { Transaction } from '../models/transaction-model';
+import { User } from '../models/user-model';
+import { BookstoreService } from './bookstore.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  private bankAccount = 1500;
+
+  constructor(private bookstoreService: BookstoreService) {}
+
+  public addMoneyToWallet(user: User): void {
+    user.wallet.currentCash.value += this.bankAccount;
+    console.log('dodano środki z Banku');
+  }
+
+  public buyBookAsGift(
```

Comment:
You should move buyBookAsGift to bookstore or buyBook to user. Keep it in one place